modules/siddhi-extensions/regex/src/test/java/org/wso2/siddhi/extension/regex/RegexPersistanceTestCase.java

```diff
@@ -62,200 +67,217 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+        QueryCallback queryCallback = new QueryCallback() {
             @Override
             public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
                 EventPrinter.print(timeStamp, inEvents, removeEvents);
                 for (Event inEvent : inEvents) {
                     count++;
                     if (count == 1) {
-                        Assert.assertEquals("WSO2 employees", inEvent.getData(1));
+                        Assert.assertEquals(false, inEvent.getData(1));
                     }
                     if (count == 2) {
-                        Assert.assertEquals("21", inEvent.getData(1));
+                        Assert.assertEquals(false, inEvent.getData(1));
                     }
                     if (count == 3) {
-                        Assert.assertEquals(" is situated in trace and its a ", inEvent.getData(1));
+                        Assert.assertEquals(true, inEvent.getData(1));
                     }
                     if (count == 4) {
-                        Assert.assertEquals(null, inEvent.getData(1));
+                        Assert.assertEquals(false, inEvent.getData(1));
                     }
                     eventArrived = true;
                 }
             }
-        });
-
+        };
+        executionPlanRuntime.addCallback("query1", queryCallback);
         InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
         executionPlanRuntime.start();
-        inputHandler.send(new Object[]{"21 products are produced within 10 years by WSO2 currently by WSO2 employees", 60.5f, "(\\d\\d)(.*)(WSO2.*)", 3});
-        inputHandler.send(new Object[]{"21 products are produced within 10 years by WSO2 currently by WSO2 employees", 60.5f, "(\\d\\d)(.*)(WSO2.*)", 1});
-        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2(.*)middleware", 1});
-        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2(.*)middleware", 2});
-        Thread.sleep(100);
+        inputHandler.send(new Object[]{null, 60.5f, "WSO2"});
+        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2"});
+        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware", 60.5f, "WSO2"});
+        Thread.sleep(500);
+        executionPlanRuntime.persist();
+        //restarting execution plan
+        executionPlanRuntime.shutdown();
+        inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+        //loading
+        executionPlanRuntime.restoreLastRevision();
+        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2"});
+        Thread.sleep(500);
         Assert.assertEquals(4, count);
         Assert.assertTrue(eventArrived);
         executionPlanRuntime.shutdown();
     }
+
```

Comment:
Please make sure all tests are running properly as you have changed existing testcases also

Reply:
In this case, I didn't change the existing testcases, i just add them to the particular files. RegexPersistanceTestCase.java  is a new class file.